src/run_metrics_on_file.ts

```diff
@@ -0,0 +1,100 @@
+import * as fs from 'fs';
+import * as net from 'net';
+import { Chess } from '../cjsmin/src/chess';
+import { gameChunks } from './fileReader';
+import {
+  KDRatioMetric,
+  KillStreakMetric,
+  MateAndAssistMetric,
+} from './metrics/captures';
+import { MoveDistanceMetric } from './metrics/distances';
+import { MetadataMetric } from './metrics/misc';
+import {
+  GameWithMostMovesMetric,
+  MiscMoveFactMetric,
+  PieceLevelMoveInfoMetric,
+} from './metrics/moves';
+import { PromotionMetric } from './metrics/promotions';
+import { RESULTS_PATH } from './queue';
+
+/**
+ *
+ * @param path
+ * @returns
+ */
+export async function main(path: string) {
+  console.time('Total Execution Time');
+  await gameIterator(path);
+  console.timeEnd('Total Execution Time');
+  return results;
+}
+
+let results = {
+  'Number of games analyzed': 0,
+};
+
+/**
+ * Metric functions will ingest a single game at a time
+ * @param metricFunctions
+ */
+async function gameIterator(path) {
+  const cjsmin = new Chess();
+
+  const gamesGenerator = gameChunks(path);
+
+  const metrics = [
+    new KDRatioMetric(),
+    new KillStreakMetric(),
+    new MateAndAssistMetric(),
+    new PromotionMetric(),
+    new MoveDistanceMetric(),
+    new GameWithMostMovesMetric(),
+    new PieceLevelMoveInfoMetric(),
+    new MetadataMetric(cjsmin),
```

Comment:
didn't realize we were passing cjsmin as an argument to MetadataMetric. why? is this necessary?

Reply:
theere's some method we are using there

Reply:
maybe we could make it static instead of a member. Not high prio for me rn